truenas/webui
# NAS-109881 / None / Added buttons around icons to make them interactable and other minor fixes for snapshots page

Added transparent `<button>` around mat-icons as per the recommendations here https://material.angular.io/components/icon/overview. The snapshots page has special cases where columns are changed on the go and some additions were also made to fix issues with that.

To test
1. Check the snapshots page and use it with a screen reader (NVDA) or just use the system with a keyboard. The icons should be intractable. (Previously, those icons would be completely skipped and not intractable without a mouse click). The icons should have proper descriptions when in focus for the screen reader via `aria-label`
2. Change the columns from the cog icon on the top right. Make sure that the columns change and the action icon on the right changes as well based on whichever configuration is selected currently. (Previously, in some cases after a reload, switching back to all columns would not show the actions menu icon on the row right)
3. Also check that the setting is persistent and UI remembers the last configuration on reload. (Previously, you would always see fewer columns selected no matter what after a reload)
4. The columns drop down trigger button at the top right previously didn't behave correctly. It wouldn't hide or show if columns config changed. That should be fixed now as well.
